### DISCUSSION TITLE
How to set timeout for ImageRequest

### DISCUSSION THREAD

**Discussion by mesudemre** [coil-kt/coil#1531] [Q&A] [answered]
Hello everyone,
Is it possible to set timeout duration to ImageRequest builder?

Thanks,

**Comment by colinrtwhite** [comment, marked as answer]
Not directly, but you can set any network properties (including request timeouts) using an OkHttp interceptor. You'll have to create an `OkHttpClient`, add the interceptor, then pass it to `ImageLoader.Builder.okHttpClient`.

**Comment by Kshitij09-sc** [comment]
@colinrtwhite  @mesudemre Just a silly doubt, how about wrapping the `imageLoader.execute(request)` coroutine inside `withTimeout { }` block?

**Reply by mesudemre** [reply]
Oh it is right. Thanks.
